<a href="https://colab.research.google.com/github/salmanmfa2/data-science-2026/blob/main/Pertemuan2_Salman_MFA_240401010356.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [67]:
import pandas as pd
import numpy as np
url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(url)



In [68]:
print(df.shape, df.columns.tolist())
print(df.isnull().sum())

(891, 12) ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


In [69]:
hasil = df.groupby('Pclass').agg(
total=('PassengerId','count'),
selamat=('Survived','mean'),).round(3)
print(hasil)

        total  selamat
Pclass                
1         216    0.630
2         184    0.473
3         491    0.242


In [70]:
survived_people = (df['Survived'] == 1).sum()
percent_survived = survived_people / df['Survived'].count() * 100
print(percent_survived)

38.38383838383838


In [71]:
wk1 = df[(df['Sex']=='female') & (df['Pclass']==1)]
print(wk1['Age'].mean())

34.61176470588235


In [72]:
hasil = df.groupby('Pclass').agg(
    total=('PassengerId','count'),
    selamat=('Survived','mean')
).round(3)
print(hasil)

hasil_only_mean = df.groupby('Pclass')['Survived'].mean()
print(hasil_only_mean)

        total  selamat
Pclass                
1         216    0.630
2         184    0.473
3         491    0.242
Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64


Pertanyaan Pertemuan 2:

1.Berapakah jumlah kolom dan baris?

Terdapat 891 Row dan 12 Column , yang didapat dari fungsi df.shape()

(Line Code 2)


2. Kolom mana yang memiliki nilai kosong?

Kolom "Age" memiliki 177 nilai kosong (null)
Kolom "Cabin" memiliki 687 nilai kosong (null)
kolom "Embarked" memiliki 2 nilai kosong (null)

(Lince Code 2)

3. Berapa persen penumpang yang selamat
(Survived=1)?

38,38% Penumpang Selamat

(Line Code 3)

4. Berapa Rata-rata usia penumpang wanita kelas 1 yang
selamat?

Rata-rata Usia 34 Tahun

(Line Code 5)

 5. Kelas penumpang mana yang tingkat keselamatan tertinggi?

 Penumpangn pada Kelas 1 memiliki tingkat keselamatan tertinggi (Pclass = 1 ; Survived = 1) dengan jumlah selamat sebanyak 216 Orang atau 63%

 (Line Code 6)

 6. Tampilkan hasil groupby('Pclass')['Survived'].mean()

(Line Code 6)
